esp-config/src/generate/mod.rs

```diff
@@ -415,60 +475,67 @@ mod test {
                     "esp-test",
                     &[
                         ConfigOption {
-                            name: "number",
-                            description: "NA",
+                            name: String::from("number"),
```

Comment:
Are these intentionally not using the new constructor?

Reply:
initially yes but testing the constructor makes sense

Reply:
@MabezDev why did you mark this resolved?

Reply:
Oops, I didn't mean to close this one.

Reply:
the existing tests still use the struct initializer, but I added a separate test for the ctor